On first startup, agent tries to remove old state file that does not exist

> 2017-03-03T02:41:51Z [INFO] Created new registry key: HKEY_LOCAL_MACHINE\SOFTWARE\Amazon\ECS Agent\State File
2017-03-03T02:41:51Z [ERROR] Error removing old file ; err remove : The system cannot find the path specified.

There's no old state file to remove, but the agent tries to remove `""` which does not exist.

There is no impact to this bug except the log line above.

--- a/agent/statemanager/state_manager_windows.go
+++ b/agent/statemanager/state_manager_windows.go
@@ -146,9 +146,14 @@ func (manager *basicStateManager) writeFile(data []byte) error {
 		seelog.Errorf("Failed to save the data file path: %v", err)
 		return err
 	}
+	if oldFile == "" {
+		return nil
+	}
+
 	err = deps.fs.Remove(oldFile)
 	if err != nil {
 		seelog.Errorf("Error removing old file %s; err %v", oldFile, err)
+		return err
 	}
 	return nil
 }

--- a/agent/statemanager/state_manager_windows_test.go
+++ b/agent/statemanager/state_manager_windows_test.go
@@ -211,4 +211,27 @@ func TestStateManagerSave(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestStateManagerNoOldStateRemoval(t *testing.T) {
+	mockRegistry, mockKey, mockFS, mockFile, manager, cleanup := setup(t)
+	defer cleanup()
+
+	basicManager := manager.(*basicStateManager)
+	gomock.InOrder(
+		mockRegistry.EXPECT().OpenKey(ecsDataFileRootKey, ecsDataFileKeyPath, gomock.Any()).Return(mockKey, nil),
+		mockKey.EXPECT().GetStringValue(ecsDataFileValueName).Return(``, uint32(0), nil),
+		mockKey.EXPECT().Close(),
+		mockFS.EXPECT().TempFile(basicManager.statePath, ecsDataFile).Return(mockFile, nil),
+		mockFile.EXPECT().Write(gomock.Any()),
+		mockFile.EXPECT().Sync(),
+		mockFile.EXPECT().Close(),
+		mockFile.EXPECT().Name().Return(`C:\new.json`),
+		mockRegistry.EXPECT().CreateKey(ecsDataFileRootKey, ecsDataFileKeyPath, gomock.Any()).Return(mockKey, false, nil),
+		mockKey.EXPECT().SetStringValue(ecsDataFileValueName, `C:\new.json`),
+		mockKey.EXPECT().Close(),
+		mockFile.EXPECT().Close(),
+	)
+	err := manager.Save()
+	assert.Nil(t, err)
+}
+
 // TODO TestStateManagerSave + errors

